Wikipedia/Code/WelcomeAnimationExtensions.swift

```diff
@@ -0,0 +1,753 @@
+import Foundation
+
+struct WelcomePlus {
+    var position: CGPoint
+    var animation: WelcomeAnimation
+}
+
+struct WelcomeCircle {
+    var position: CGPoint
+    var radius: CGFloat
+    var isDashed: Bool
+    var animation: WelcomeAnimation
+}
+
+struct WelcomeLine {
+    var position: CGPoint
+    var width: CGFloat
+    var animation: WelcomeAnimation
+}
+
+struct WelcomeImage {
+    var name: String
+    var frame: CGRect
+    var zPosition: CGFloat
+    var animation: WelcomeAnimation
+}
+
+struct WelcomeBar {
+    var frame: CGRect
+    var color: UIColor
+    var cornerRadius: CGFloat
+    var animation: WelcomeAnimation
+}
+
+struct WelcomeAnimation {
+    var delay: Double
+    var duration: Double
+    var initialOpacity: Double
+    var finalOpacity: Double
+    var initialTransform: CATransform3D
+    var finalTransform: CATransform3D
+}
+
+extension UIBezierPath {
+    class func wmf_circlePathWithRadius(radius: CGFloat) -> UIBezierPath {
+        return UIBezierPath(
+            arcCenter: CGPointZero,
+            radius: radius,
+            startAngle: 0.0,
+            endAngle: CGFloat(M_PI * 2.0),
+            clockwise: true
+        )
+    }
+    class func wmf_horizontalLinePathWithWidth(width: CGFloat) -> UIBezierPath {
+        let path = UIBezierPath()
+        path.moveToPoint(CGPointMake(width * -0.5, 0.0))
+        path.addLineToPoint(CGPointMake(width * 0.5, 0.0))
+        return path
+    }
+    class func wmf_verticalLinePathWithHeight(height: CGFloat) -> UIBezierPath {
+        let path = UIBezierPath()
+        path.moveToPoint(CGPointMake(0.0, height * -0.5))
+        path.addLineToPoint(CGPointMake(0.0, height * 0.5))
+        return path
+    }
+    class func wmf_plusPathWithWidth(width: CGFloat) -> UIBezierPath {
+        let path = UIBezierPath.wmf_horizontalLinePathWithWidth(width)
+        path.appendPath(UIBezierPath.wmf_verticalLinePathWithHeight(width))
+        return path
+    }
+}
+
+extension CAShapeLayer {
+    func wmf_applyCommonSettingsForSize(size: CGSize, position: CGPoint, initialOpacity: Double) -> Void {
+        self.lineCap = "round"
+        self.opacity = Float(initialOpacity)
+        self.lineWidth = size.width * 0.014
+        self.strokeEnd = 1.0
+        self.position = CGPointMake(size.width * position.x, size.height * position.y)
+        self.zPosition = 100
+    }
+    class func wmf_addLine(line: WelcomeLine, toView: UIView) -> CAShapeLayer {
```

Comment:
This seems to be more than adding a line. Also its returning a shape layer… so this seems more like a constructor `shapeLayerWithLine…` or something to that affect